tests/desktop/developer_hub/test_developer_hub_submit_apps.py

```diff
@@ -233,3 +235,80 @@ def test_hosted_app_submission(self, mozwebqa_devhub_logged_in):
+    def test_check_submission_of_an_app_with_XSS_in_its_app_name(self, mozwebqa_devhub_logged_in):
+        if '-dev.allizom' in mozwebqa_devhub_logged_in.base_url:
+            env = 'dev'
+        else:
+            env = 'stage'
+
+        app = MockApplication(env, app_type='xss_app')
+
+        dev_home = Home(mozwebqa_devhub_logged_in)
+
+        dev_agreement = dev_home.click_submit_new_app()
+
+        """Agree with the developer agreement and continue if it was not accepted
+        in a previous app submit"""
+        manifest_validation_form = dev_agreement.click_continue()
+
+        # select device type
+        for device in app['device_type']:
+            if device[1]:
+                manifest_validation_form.device_type(device[0])
+
+        manifest_validation_form.app_type(app['app_type'])
+
+        # submit the app manifest url and validate it
+        manifest_validation_form.type_app_manifest_url(app['url'])
+        manifest_validation_form.click_validate()
+        Assert.true(manifest_validation_form.app_validation_status,
+                    msg=manifest_validation_form.app_validation_message)
+        try:
+            app_details = manifest_validation_form.click_continue()
+            Assert.true(app_details.is_the_current_submission_stage, '\n Expected step is: Details \n Actual step is: %s' % app_details.current_step)
+
+            # add custom app details for every field
+            app_details.click_change_name()
+            app_details.type_url_end(app['url_end'])
+            app_details.type_description(app['description'])
+            app_details.type_privacy_policy(app['privacy_policy'])
+            app_details.type_homepage(app['homepage'])
+            app_details.type_support_url(app['support_website'])
+            app_details.type_support_email(app['support_email'])
+
+            for category in app['categories']:
+                # check/uncheck the checkbox according to the app value
+                app_details.select_categories(*category)
+
+            app_details.screenshot_upload(app['screenshot_link'])
+
+            next_steps = app_details.click_continue()
+            Assert.equal('Almost There!', next_steps.almost_there_message)
+
+            content_ratings = next_steps.click_continue()
+            Assert.equal('Get My App Rated', content_ratings.get_app_rated_message)
+
+            # insert Submission ID and Security code to get app rated
+            content_ratings.fill_in_app_already_rated_info(app['submission_id'], app['security_code'])
+            content_ratings.click_submit()
+            Assert.equal('Congratulations, your app submission is now complete and will be reviewed shortly!',
+                             content_ratings.saved_ratings_message)
+
+            # check that xss is in app name
+            edit_listing_page = dev_home.go_to_edit_listing_page(app)
+            Assert.contains(u"<script>alert(‘XSS')</script>", edit_listing_page.page_title)
```

Comment:
We are checking that the xss is in the app name, but is this sufficient to ensure that it's not being executed? I am guessing so, because if an alert popped up that would interfere with the test, but I wanted to ask the question.